croptalk/tools.py

```diff
@@ -1,52 +1,111 @@
+    Words: {question}
+    commodity: """
 
-    Returns: the tool returns a string which gives the coverage level with their
-            associated percentage indemnified statistic
+    commodity_prompt = PromptTemplate.from_template(commodity_template_tool)
+    return commodity_prompt | llm | StrOutputParser()
 
+
+@tool("get_SP_doc")
+def get_sp_document(state: Optional[str] = None,
+                    county: Optional[str] = None,
+                    commodity: Optional[str] = None,
+                    year: Optional[int] = None):
     """
+    This tool is used to query the SP document from a database along state, county, commodity and year.
+    For instance, a user might ask :
+    - find me the SP document related to Oranges in Yakima, Washington for the year 2024
+    - SP document for corn in Butte, California, 2022
 
-    sob = SOB[SOB["commodity_year"] == 2023]
+    Args:
+    state : name of state
+    county_name: name of county provided
+    commodity_name: name of commodity
+    insurance_plan_name: provided plan name abbreviation
 
-    sob = sob[
-        (sob["state_abbreviation"] == state_abbreviation) &
-        (sob["county_name"] == county_name) &
-        (sob["commodity_name"] == commodity_name.lower()) &
-        (sob["insurance_plan_name_abbreviation"] == insurance_plan_name)
-        ]
+    Returns: str
+    """
 
-    return "In 2023, we observe the following data" + str(
-        [{"coverage_level": i, metric: j} for i, j in zip(list(sob["coverage_level"]), list(sob[metric]))]
-    )
+    if all([state, county, commodity, year]):
+        # use chain to retrieve correct commodity info
```

Comment:
Not sure why commodity is provided, but we still need to invoke commodity_chain... Maybe worth adding a comment as to why this is required? And why it is required for commodity, but not for state and county?

Reply:
good point, I'll add a comment.

Overall, it's because the tool uses the argument descriptions to extract them from the input. In the case of commodities, there will be spelling differences which will not be corrected (blueberry -> blueberries)